manticore/core/state.py

```diff
@@ -461,20 +461,31 @@ def solve_one_n(self, *exprs, constrain=False):
         :return: Concrete value or a tuple of concrete values
         :rtype: int
         """
-        values = []
-        for expr in exprs:
-            if not issymbolic(expr):
-                values.append(expr)
-            else:
-                expr = self.migrate_expression(expr)
-                value = self._solver.get_value(self._constraints, expr)
-                if constrain:
-                    self.constrain(expr == value)
-                # Include forgiveness here
-                if isinstance(value, bytearray):
-                    value = bytes(value)
-                values.append(value)
-        return values
+        return self.solve_one_n_batched(exprs, constrain)
+
+    def solve_one_n_batched(self, exprs, constrain=False):
+        """
+        Concretize a symbolic :class:`~manticore.core.smtlib.expression.Expression` into
+        one solution.
+        :param exprs: An iterable of manticore.core.smtlib.Expression
+        :param bool constrain: If True, constrain expr to solved solution value
+        :return: Concrete value or a tuple of concrete values
+        :rtype: int
```

Comment:
I think this is done.

Reply:
uhm, I'm getting:

> manticore/core/state.py:466: error: Argument 1 to "solve_one_n_batched" of "StateBase" has incompatible type "Tuple[List[Expression], ...]"; expected "List[Expression]"

from mypy. What is the correct type for `*exprs` in solve_one_n:

```
    def solve_one_n(self, *exprs: List[Expression], constrain: bool = False) -> List[int]:
        """
        Concretize a list of symbolic :class:`~manticore.core.smtlib.expression.Expression` into
        a list of solutions.

        :param exprs: An iterable of manticore.core.smtlib.Expression
        :param bool constrain: If True, constrain expr to solved solution value
        :return: List of concrete value or a tuple of concrete values
        """
        return self.solve_one_n_batched(exprs, constrain)
```

Reply:
I'm not 100% certain which behavior you're intending since that mypy traceback has `*exprs` as an argument whereas the version I see on Github just has `exprs`. I'm going to assume it's the version in the mypy traceback because this function seems intended to replace `solve_one_n`, which also has that signature. In that case, this argument will indeed be a tuple containing list(s) of arguments, so you should be able to update the type signature to the one mypy prints out.